### PR TITLE
fix: genesis type cast

### PR DESCRIPTION
**Pull request by kosegor**
## Description

closes: #XXXX

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [x] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [x] added `!` to the type prefix if API or client breaking change
- [x] added appropriate labels to the PR
- [x] targeted the correct branch (see [PR Targeting](https://github.com/umee-network/umee/blob/main/CONTRIBUTING.md#pr-targeting))
- [x] provided a link to the relevant issue or specification
- [x] added a changelog entry to `CHANGELOG.md`
- [x] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [x] updated the relevant documentation or specification
- [x] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items._

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -57,6 +57,7 @@ Ref: https://keepachangelog.com/en/1.0.0/
 - [1956](https://github.com/umee-network/umee/pull/1956) Leverage: token liquidation threshold must be bigger than collateral_weight.
 - [1954](https://github.com/umee-network/umee/pull/1954) Leverage: `MaxBorrow` now accounts for
   `MinCollateralLiquidity` and `MaxSupplyUtilization`
+- [1968](https://github.com/umee-network/umee/pull/1968) Leverage: genesis type cast.
```
Comment (toteki) [inline CHANGELOG.md:60]:
```suggestion
- [1968](https://github.com/umee-network/umee/pull/1968) Leverage: fix type cast of AdjustedBorrow in ExportGenesis
```

**Review by toteki — APPROVED** [review]
Looks good - thanks for catching this fix

File: x/leverage/keeper/genesis.go [hunk 2]
```diff
@@ -70,13 +70,13 @@ func (k Keeper) getAllAdjustedBorrows(ctx sdk.Context) []types.AdjustedBorrow {
 		addr := types.AddressFromKey(key, prefix)
 		denom := types.DenomFromKeyWithAddress(key, prefix)
 
-		var amount sdkmath.Int
+		var amount sdk.Dec
```
Comment (robert-zaremba) [inline x/leverage/keeper/genesis.go:73]: oh, we had a bug here?

**Review by robert-zaremba — APPROVED** [review]

File: x/leverage/keeper/genesis.go (hunk 2 above)
Comment (kosegor) [reply x/leverage/keeper/genesis.go:73]: yes sir

**Pull request merged**